### PR TITLE
For  AWS Lambda default to Micronaut HTTP Client JDK instead of netty http client

### LINKED ISSUE #2271: for lambda default to Micronaut HTTP Client JDK instead of netty http client
This may impact in startup performance.

### PR DESCRIPTION
**Pull request by wetted**
closes #2271

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/awslambdacustomruntime/AwsLambdaCustomRuntime.java [hunk 1]
```diff
@@ -64,8 +64,8 @@ public void processSelectedFeatures(FeatureContext featureContext) {
         if (awsLambda.supports(featureContext.getApplicationType()) && !featureContext.isPresent(AwsLambda.class)) {
             featureContext.addFeature(awsLambda);
         }
-        if (!featureContext.isPresent(HttpClient.class)) {
-            featureContext.addFeature(httpClient);
+        if (!featureContext.isPresent(HttpClientJdk.class)) {
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/awslambdacustomruntime/AwsLambdaCustomRuntime.java]: this should be: 

```suggestion
        if (!featureContext.isPresent(HttpClientFeature.class)) {
```?

File: starter-core/src/main/java/io/micronaut/starter/feature/awslambdacustomruntime/AwsLambdaCustomRuntime.java [hunk 2]
```diff
@@ -51,11 +51,11 @@ public class AwsLambdaCustomRuntime implements FunctionFeature, ApplicationFeatu
     public static final String FEATURE_NAME_AWS_LAMBDA_CUSTOM_RUNTIME = "aws-lambda-custom-runtime";
 
     private final Provider<AwsLambda> awsLambda;
-    private final HttpClient httpClient;
+    private final HttpClientJdk httpClientJdk;
 
-    public AwsLambdaCustomRuntime(Provider<AwsLambda> awsLambda, HttpClient httpClient) {
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/awslambdacustomruntime/AwsLambdaCustomRuntime.java:56]: we should deprecate the old constructor

File: starter-core/src/main/java/io/micronaut/starter/feature/awslambdacustomruntime/AwsLambdaCustomRuntime.java (hunk 1 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/awslambdacustomruntime/AwsLambdaCustomRuntime.java]: resolved via https://github.com/micronaut-projects/micronaut-starter/pull/2272/commits/cfd040e1ab589e6c8431163a4300e1b355768f3c

File: starter-core/src/main/java/io/micronaut/starter/feature/awslambdacustomruntime/AwsLambdaCustomRuntime.java (hunk 2 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/awslambdacustomruntime/AwsLambdaCustomRuntime.java:56]: resolved via https://github.com/micronaut-projects/micronaut-starter/pull/2272/commits/a66e38845727bcae3f66de931512b8cf98a85c85 and https://github.com/micronaut-projects/micronaut-starter/pull/2272/commits/07b6e0db5f48cb220467f88bad443b0ff2081e5e

**Review by sdelamo — APPROVED** [review]

**Review by timyates — APPROVED** [review]

**Pull request merged**